src/gt4py/next/ffront/ast_passes/single_static_assign.py

```diff
@@ -11,12 +11,77 @@
 # distribution for a copy of the license or check <https://www.gnu.org/licenses/>.
 #
 # SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import annotations
 
 import ast
+import dataclasses
+import typing
 
 from gt4py.next.ffront.fbuiltins import TYPE_BUILTIN_NAMES
 
 
+ASTNodeT = typing.TypeVar("ASTNodeT", bound=ast.AST)
+
+_UNIQUE_NAME_SEPERATOR = "ᐞ"
+
+
+def unique_name(name: str, num_assignments: int):
```

Comment:
Missing return type annotation. Additionally, is this the right place to add this function? It is used in many different tests but I understand it might be out of the scope of this PR, so I suggest to add at least a TODO mentioning it.